plugins/reach/lib/model/EntryVendorTask.php

```diff
@@ -453,49 +453,68 @@ public function postSave(PropelPDO $con = null)
 
 	public function addSchedulingData()
 	{
+		$language = '';
+		$feature = null;
+		/* @var $taskData kScheduledVendorTaskData */
+		$taskData = $this->getTaskJobData();
+		/* @var $connectedEvent LiveStreamScheduleEvent */
+		$connectedEvent = $taskData->getScheduleEvent();
+		/* @var $vendorCatalogItem VendorCatalogItem */
+		$vendorCatalogItem = VendorCatalogItemPeer::retrieveByPK($this->getCatalogItemId());
+
 		switch ($this->getServiceFeature())
 		{
+			case KalturaVendorServiceFeature::LIVE_TRANSLATION:
+			{
+				$language = languageCodeManager::getThreeCodeFromKalturaName($vendorCatalogItem->getTargetLanguage());
```

Comment:
since either way we set the language in addLanguageToFeature i think we can move the language extraction to inside the function and just pass to it the source/target language

Reply:
in that case, the switch moves to the helper function, while in the future we might need to separate between other features. The division, in my eyes, will be blurry again.